tools/perfscale-load-generator/executor/vmi.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 IBM, Inc.
+ *
+ */
+
+package executor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/time/rate"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	kvv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/tests"
+
+	"kubevirt.io/kubevirt/tools/perfscale-load-generator/config"
+	"kubevirt.io/kubevirt/tools/perfscale-load-generator/flags"
+)
+
+const userData string = "#!/bin/bash\necho 'hello'\n"
+
+// CreateVMISpecWithResources create a new VMI spec with random name and ephemeral disk, and with the given resource and namespace
+func CreateVMISpecWithResources(namespace string, spec config.VMISpec) *kvv1.VirtualMachineInstance {
+	imgName := fmt.Sprintf("%s/%s-container-disk-demo:%s", flags.VMIImgRepo, spec.VMImage, flags.VMIImgTag)
```

Comment:
I wouldn't do templating on the Image Name here, just let the user provide any image they want and do sane defaults in the start.